src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrItemResource.java

```diff
@@ -135,6 +139,21 @@ protected String getResourceTypeForNode(final Node node)
         return result;
     }
 
+    /**
+     * @return Parent node or null if not existing or root
+     */
+    public @Nullable Node getParentNode() {
+        String parentPath = ResourceUtil.getParent(getPath());
+        if (StringUtils.isNotBlank(parentPath)) {
+            try {
+                return (Node)((JackrabbitSession)item.getSession()).getItemOrNull(parentPath);
```

Comment:
this code should check whether session is a JackrabbitSession and if not just use getItem